ceno_rt/src/syscalls.rs

```diff
@@ -1,18 +1,18 @@
 // Based on https://github.com/succinctlabs/sp1/blob/013c24ea2fa15a0e7ed94f7d11a7ada4baa39ab9/crates/zkvm/entrypoint/src/syscalls/keccak_permute.rs
-
-const KECCAK_PERMUTE: u32 = 0x00_01_01_09;
-
+#[cfg(target_os = "zkvm")]
 use core::arch::asm;
 
+pub const KECCAK_PERMUTE: u32 = 0x00_01_01_09;
+
 /// Executes the Keccak256 permutation on the given state.
 ///
 /// ### Safety
 ///
 /// The caller must ensure that `state` is valid pointer to data that is aligned along a four
 /// byte boundary.
 #[allow(unused_variables)]
-#[no_mangle]
-pub extern "C" fn syscall_keccak_permute(state: &mut [u64; 25]) {
+pub fn keccak_permute(state: &mut [u64; 25]) {
```

Comment:
No need to prefix the name of function with the name of module.  Just refer to the function as `syscalls::keccak_permute` to get the same effect.